manager/src/test/java/com/redhat/service/smartevents/manager/connectors/ConnectorsServiceTest.java

```diff
@@ -33,77 +44,85 @@ class ConnectorsServiceTest {
+        verify(connectorsDAOMock).persist(captor.capture());
+        ConnectorEntity entity = captor.getValue();
+        assertThat(entity.getConnectorType()).isEqualTo(expectedConnectorType);
     }
 
     @Test
     @Transactional
     void doNotDeleteConnector() {
         when(connectorsDAOMock.findByProcessorId(TEST_PROCESSOR_ID)).thenReturn(null);
-
-        connectorsService.deleteConnectorEntity(testProcessor());
-
+        connectorsService.deleteConnectorEntity(testProcessor(testWebhookAction()));
         verify(connectorsDAOMock, never()).delete(any(ConnectorEntity.class));
     }
 
-    @Test
+    @ParameterizedTest
     @Transactional
-    void doDeleteConnector() {
-        when(connectorsDAOMock.findByProcessorId(TEST_PROCESSOR_ID)).thenReturn(testConnectorEntity());
-
-        connectorsService.deleteConnectorEntity(testProcessor());
-
+    @MethodSource("connectorProcessors")
+    void doDeleteConnector(Processor processor) {
+        when(connectorsDAOMock.findByProcessorId(TEST_PROCESSOR_ID)).thenReturn(testConnectorEntity(processor));
+        connectorsService.deleteConnectorEntity(processor);
         verify(connectorsDAOMock, never()).delete(any(ConnectorEntity.class));
     }
 
-    private ConnectorEntity testConnectorEntity() {
+    private static Stream<Arguments> connectorProcessors() {
+        Object[][] arguments = {
+                { testProcessor(testSlackAction()), SlackActionConnector.CONNECTOR_TYPE },
+                { testProcessor(testSlackSource()), SlackSourceConnector.CONNECTOR_TYPE }
+        };
+        return Stream.of(arguments).map(Arguments::of);
+    }
+
+    private static ConnectorEntity testConnectorEntity(Processor processor) {
         ConnectorEntity connectorEntity = new ConnectorEntity();
         connectorEntity.setId(TEST_CONNECTOR_ID);
         connectorEntity.setConnectorExternalId(TEST_CONNECTOR_EXTERNAL_ID);
-        connectorEntity.setProcessor(testProcessor());
+        connectorEntity.setProcessor(processor);
         return connectorEntity;
     }
 
-    private Processor testProcessor() {
+    private static Processor testProcessor(Gateway gateway) {
         Processor processor = new Processor();
-        processor.setType(ProcessorType.SINK);
+
+        ProcessorDefinition processorDefinition = new ProcessorDefinition();
+        if (gateway instanceof Action) {
+            processorDefinition.setRequestedAction((Action) gateway);
+            processor.setType(ProcessorType.SINK);
+        } else {
+            processorDefinition.setRequestedSource((Source) gateway);
+            processor.setType(ProcessorType.SOURCE);
+        }
+
         processor.setId(TEST_PROCESSOR_ID);
         processor.setName(TEST_PROCESSOR_NAME);
+        processor.setDefinition(processorDefinition);
         return processor;
     }
 
-    private Action testSlackAction() {
+    private static Action testSlackAction() {
```

Comment:
Minor note (and it is already on main so not introduced by this PR): I find confusing having a method named `test` if it is not a test :)

Reply:
I renamed those methods to avoid confusion :+1: